sdk/core/azure-core-test/src/main/java/com/azure/core/test/utils/TestProxyDownloader.java

```diff
@@ -0,0 +1,202 @@
+                    }
+                }
+                entry = archive.getNextEntry();
+            }
+        } catch (IOException | ArchiveException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static File getOutputFile(ArchiveEntry entry) {
+        return new File(PROXYPATH.toFile(), entry.getName());
+    }
+
+
+    private static void downloadProxy(PlatformInfo platformInfo) {
+        LOGGER.log(LogLevel.INFORMATIONAL, () -> "Downloading test proxy. This may take a few moments.");
+
+        try {
+            URL url = new URL(getProxyDownloadUrl(platformInfo));
+            Files.copy(url.openStream(),
+                getZipFileLocation(platformInfo.getExtension()),
+                StandardCopyOption.REPLACE_EXISTING);
+        } catch (IOException e) {
+            throw new RuntimeException("Could not save test proxy download", e);
+        }
+
+        updateDownloadedFileVersion();
+    }
+
+    private static Path getZipFileLocation(String extension) {
+        return Paths.get(System.getProperty("java.io.tmpdir"), String.format("testproxy.%s", extension));
+    }
+
+    private static void updateDownloadedFileVersion() {
+        Path filePath = getFileVersionPath();
+        try {
+            Files.write(filePath, Arrays.asList(TEST_PROXY_TAG));
+        } catch (IOException e) {
+            throw new RuntimeException("Failed to write version data to file", e);
+        }
+    }
+
+    private static boolean checkDownloadedVersion() {
+        Path filePath = getFileVersionPath();
+        if (!Files.exists(filePath)) {
+            return false;
+        }
+        String fileVersion;
+        try {
+            fileVersion = Files.readAllLines(filePath).get(0);
+        } catch (IOException e) {
+            throw new RuntimeException("Failed to read version data from file", e);
+        }
+        return fileVersion.equals(TEST_PROXY_TAG);
+    }
+
+    private static Path getFileVersionPath() {
+        return Paths.get(System.getProperty("java.io.tmpdir"), "test-proxy-version.txt");
+    }
+
+    private static String getProxyDownloadUrl(PlatformInfo platformInfo) {
+        return String.format("https://github.com/Azure/azure-sdk-tools/releases/download/%s/test-proxy-standalone-%s-%s.%s",
+            TEST_PROXY_TAG,
+            platformInfo.getPlatform(),
+            platformInfo.getArchitecture(),
+            platformInfo.getExtension());
+    }
+
+    private static class PlatformInfo {
+        private final String platform;
+        private final String extension;
+        private final String architecture;
+
+
+        PlatformInfo() {
+            String osName = System.getProperty("os.name").toLowerCase(Locale.ROOT);
+            if (osName.contains("windows")) {
+                platform = "win";
+                extension = "zip";
+            } else if (osName.contains("linux")) {
+                platform = "linux";
```

Comment:
Would using linux as the default be safer than macOs? Changing this else if for macOs and the else for Linux

Reply:
Good callout. I just hadn't grabbed my Mac from the other room the day I wrote this to validate what the `osName` is. :) Updated this to not have a default, which is what I really want.